lib/matplotlib/image.py

```diff
@@ -1004,106 +904,30 @@ def __init__(self, fig,
 
         kwargs are an optional list of Artist keyword args
         """
-        martist.Artist.__init__(self)
-        cm.ScalarMappable.__init__(self, norm, cmap)
-        if origin is None:
-            origin = rcParams['image.origin']
-        self.origin = origin
+        _ImageBase.__init__(
+            self, None, norm=norm, cmap=cmap, origin=origin)
         self.figure = fig
         self.ox = offsetx
         self.oy = offsety
         self.update(kwargs)
         self.magnification = 1.0
 
-    def contains(self, mouseevent):
-        """Test whether the mouse event occured within the image."""
-        if six.callable(self._contains):
-            return self._contains(self, mouseevent)
-        xmin, xmax, ymin, ymax = self.get_extent()
-        xdata, ydata = mouseevent.x, mouseevent.y
-
-        if xdata is not None and ydata is not None:
-            inside = ((xdata >= xmin) and (xdata <= xmax) and
-                      (ydata >= ymin) and (ydata <= ymax))
-        else:
-            inside = False
-
-        return inside, {}
```

Comment:
It was copy-and-pasted from `AxesImage`.  Now that all images inherit from `_ImageBase`, there is a single `contains` implementation for all images.